Simulation and Modeling Lab  
**Course Code:** CSE 413  
**Academic Level/Term:** 4/1  
**Credit Hour:** 3  
**Course Teacher:** Audity Ghosh, Lecturer, Department of Computer Science and Engineering, University of Information Technology and Sciences (UITS)  
**Date:** 29 July, 2025  

\
**Student Name**: Reyazul Islam

**Student ID**: 0432320005101146

# Necessary Libraries

In [1]:
import numpy as np
from sympy import Matrix
import matplotlib.pyplot as plt
import seaborn as sns


# Matrix Functions

In [2]:
d1=6
d2=4

A = np.array([[d1+2, d2+1], [2*d1, d2+2]])

# 1. Size of a matrix
size_A = A.shape  # Returns dimensions (rows, columns)

# 2. Determinant of a matrix
det_A = np.linalg.det(A)  # Returns the determinant of the matrix

# 3. Inverse of a matrix
if np.linalg.det(A) != 0:  # Check if matrix is invertible
    inv_A = np.linalg.inv(A)  # Compute the inverse of A
else:
    inv_A = "Matrix is singular, cannot compute inverse"  # No inverse if determinant is 0

# 4. Rank of a matrix
rank_A = np.linalg.matrix_rank(A)  # Returns the rank of the matrix

# 5. Eigenvalues and eigenvectors
eigenvalues, eigenvectors = np.linalg.eig(A)  # Compute eigenvalues and eigenvectors


# Display results
print("Matrix A:\n", A)
print("Size of A:", size_A)
print("Determinant of A:", det_A)
print("Inverse of A:\n", inv_A)
print("Rank of A:", rank_A)

print("Eigenvalues of A:", eigenvalues)
print("Eigenvectors of A:\n", eigenvectors)



Matrix A:
 [[ 8  5]
 [12  6]]
Size of A: (2, 2)
Determinant of A: -12.0
Inverse of A:
 [[-0.5         0.41666667]
 [ 1.         -0.66666667]]
Rank of A: 2
Eigenvalues of A: [14.81024968 -0.81024968]
Eigenvectors of A:
 [[ 0.5918113  -0.49357476]
 [ 0.80607654  0.86970337]]


In [3]:
d1=6
d2=4

#changed the value of second row second column to d1+2 to d1+3

A = np.array([[d1+2, d2+1], [2*d1, d2+3]])

# 1. Size of a matrix
size_A = A.shape  # Returns dimensions (rows, columns)

# 2. Determinant of a matrix
det_A = np.linalg.det(A)  # Returns the determinant of the matrix

# 3. Inverse of a matrix
if np.linalg.det(A) != 0:  # Check if matrix is invertible
    inv_A = np.linalg.inv(A)  # Compute the inverse of A
else:
    inv_A = "Matrix is singular, cannot compute inverse"  # No inverse if determinant is 0

# 4. Rank of a matrix
rank_A = np.linalg.matrix_rank(A)  # Returns the rank of the matrix

# 5. Eigenvalues and eigenvectors
eigenvalues, eigenvectors = np.linalg.eig(A)  # Compute eigenvalues and eigenvectors


# Display results
print("Matrix A:\n", A)
print("Size of A:", size_A)
print("Determinant of A:", det_A)
print("Inverse of A:\n", inv_A)
print("Rank of A:", rank_A)

print("Eigenvalues of A:", eigenvalues)
print("Eigenvectors of A:\n", eigenvectors)



Matrix A:
 [[ 8  5]
 [12  7]]
Size of A: (2, 2)
Determinant of A: -4.000000000000007
Inverse of A:
 [[-1.75  1.25]
 [ 3.   -2.  ]]
Rank of A: 2
Eigenvalues of A: [15.26208735 -0.26208735]
Eigenvectors of A:
 [[ 0.56709206 -0.51774682]
 [ 0.82365442  0.85553389]]


Comparison & Explanation

1.How did the determinant change and why?

Ans.

Determinant of A = -12, B = -4

The determinant changed from −12 to −4 .

For A: (8)\*(6)−(5)\*(12) = 48−60 = −12

For B: (8)\*(7)−(5)\*(12) = 56−60 = −4

The change occurred because the bottom-right value changed from 6 to 7.

2.Did the rank change? Explain.

Ans.

No, both matrices have rank 2.

Since the determinant of both matrices is non-zero, both matrices are full rank for a 2×2 matrix.

Therefore: Rank(A) = Rank(B) = 2

3.How did the eigenvalues respond to the value change?

Ans.

For A: Eigenvalues ≈ 14.81, −0.81

For B: EIgenvalues ≈ 15.26, −0.26

So, after changing 6 to 7:

The larger eigenvalue increased: 14.81 to 15.26
The negative eigenvalue moved closer to zero: −0.81 to −0.26




4.Is B easier or harder to invert than A? Why?

Ans.

B is easier to invert than A because its determinant is further from 0:

A: det = −4
B: det = −12

So, B is further from singularity and more stable to invert.